### DISCUSSION TITLE
Chakra CLI postinstall

### DISCUSSION THREAD

**Discussion by vlbin** [chakra-ui/chakra-ui#6099] [Q&A] [answered]
Hi!
I'm building an extended component library, built on Chakra UI. I have created some custom variants and colors which I want to have typed. I use the @chakra-ui/cli as documented and it all works in my component library directory, both directly in the terminal and via a build script in my package.json. If I look in the `node_modules/@chakra-ui/styled-system/dist/declarations/src/theming.types.d.ts` file my custom variants are there, perfect!

However, when I install my component library into my React application it should run the Chakra CLI via postinstall/prepare (I've tried both). It runs without errors but it does not change the themings file and I don't get IntelliSense for my custom variants. This is my script:

```
"postinstall": "chakra-cli tokens lib/theme/index.ts && rollup -c"
```

Does anyone have a good way of going about this?

**Comment by stevaras** [comment]
@vlbin Did you have any luck on that??

**Reply by vlbin** [reply]
@stevaras I did some explaining here, #4097. is that helpful?

**Comment by stevaras** [comment, marked as answer]
> @stevaras I did some explaining here, #4097. is that helpful?

Yes it is. Although I was coming here to tell you that I managed to make it work. 
Probably yours is cleaner and more sophisticated, but what I ended up doing is : 

        "gen:theme-typings": "chakra-cli tokens ./dist/index.cjs.js --out ../../@chakra-ui/styled-system/dist/theming.types.d.ts",

This seems to make it work in the consumer ( of my custom theme ) app :D 

But thanks for also replying. I'll might take a deeper look at some point to your suggestion/solution <3 

Thanks :)

**Reply by vlbin** [reply]
Yours does indeed seem cleaner, nice!
